test/dynamo/test_misc.py

```diff
@@ -574,6 +574,25 @@ def f(a, mode):
         cleanup_op("mylib::foo")
         del lib
 
+    def test_auto_functionalize_can_with_none_return(self):
+        lib = torch.library.Library("mylib", "FRAGMENT")
+        lib.define("foo(Tensor x, Tensor(a!) out) -> None")
```

Comment:
use torch.library._scoped_library

Reply:
oh, just saw it. let me fix it in next commit.

Reply:
bonus points if you refactor the other auto_functionalized tests to use scoped_library too